home/home_7.1/main.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+
+enum Song: String {
+    case intro = "Intro"
+    case starman = "Starman"
+    case showMustGoOn = "Show must go on"
+    case letItBe = "Let it be"
+    case inTheEnd = "But in the end, it doesn't even matter"
+}
+
+enum Command: String {
+    case sad, fun, silly, dangerous
+}
+
+class MusicPlayer {
+    private var currentSong: Song = .intro
+    
+    init() {
+        print("Now playing: \(currentSong.rawValue)")
+    }
+    
+    func handleCommand(_ command: Command) {
+        switch (currentSong, command) {
```

Comment:
Let's split logic between different states to encapsulate the logic of switches from current state. That is the idea of state machine - let's suppose that we are at state S then we can easily understand the behavior of the system just by looking handlers inside corresponding class or function